gameoflife/src/test/kotlin/com/anaplan/engineering/kazuki/gameoflife/TestConway.kt

```diff
@@ -0,0 +1,303 @@
+
+import com.anaplan.engineering.kazuki.core.PreconditionFailure
+import com.anaplan.engineering.kazuki.core.dunion
+import com.anaplan.engineering.kazuki.core.minus
+import com.anaplan.engineering.kazuki.core.set
+import com.anaplan.engineering.kazuki.gameoflife.Conway.around
+import com.anaplan.engineering.kazuki.gameoflife.Conway.deadCells
+import com.anaplan.engineering.kazuki.gameoflife.Conway.disappearN
+import com.anaplan.engineering.kazuki.gameoflife.Conway.disappearNP
+import com.anaplan.engineering.kazuki.gameoflife.Conway.generation
+import com.anaplan.engineering.kazuki.gameoflife.Conway.generations
+import com.anaplan.engineering.kazuki.gameoflife.Conway.gliderN
+import com.anaplan.engineering.kazuki.gameoflife.Conway.gliderNP
+import com.anaplan.engineering.kazuki.gameoflife.Conway.isOffset
+import com.anaplan.engineering.kazuki.gameoflife.Conway.neighbourCount
+import com.anaplan.engineering.kazuki.gameoflife.Conway.newCells
+import com.anaplan.engineering.kazuki.gameoflife.Conway.offset
+import com.anaplan.engineering.kazuki.gameoflife.Conway.periodN
+import com.anaplan.engineering.kazuki.gameoflife.Conway.periodNP
+import com.anaplan.engineering.kazuki.gameoflife.Conway_Module.mk_Point
+import com.anaplan.engineering.kazuki.gameoflife.Conway_Module.mk_Population
+import kotlin.test.Ignore
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertFailsWith
+
+
+class TestConway {
+
+    private val block = mk_Population(mk_Point(0, 0), mk_Point(-1, 0), mk_Point(0, -1), mk_Point(-1, -1))
+
+    private val blinker = mk_Population(mk_Point(-1, 0), mk_Point(0, 0), mk_Point(1, 0))
+
+    private val toad = mk_Population(
+        mk_Point(-1, 0), mk_Point(0, 0), mk_Point(1, 0),
+        mk_Point(0, -1), mk_Point(-1, -1), mk_Point(-2, -1)
+    )
+
+    private val beacon = mk_Population(
+        mk_Point(-2, 0), mk_Point(-2, 1), mk_Point(-1, 1), mk_Point(0, -2),
+        mk_Point(1, -2), mk_Point(1, -1)
+    )
+
+    private val pQuad = mk_Population(
+        mk_Point(2, 1), mk_Point(3, 1), mk_Point(3, 2),
+        mk_Point(1, 2), mk_Point(1, 3), mk_Point(2, 3), mk_Point(5, 2),
+        mk_Point(5, 3), mk_Point(6, 3), mk_Point(7, 3), mk_Point(2, 5),
+        mk_Point(3, 5), mk_Point(3, 6), mk_Point(3, 7)
+    )
+
+    private val pulsar = mk_Population(
+        dunion(
+            set(pQuad) { point -> point },
+            set(pQuad) { point -> mk_Point(-point.x, point.y) },
+            set(pQuad) { point -> mk_Point(point.x, -point.y) },
+            set(pQuad) { point -> mk_Point(-point.x, -point.y) }
+        ))
+
+    private val diehard = mk_Population(
+        mk_Point(0, 1), mk_Point(1, 1), mk_Point(1, 0), mk_Point(0, 5),
+        mk_Point(0, 6), mk_Point(0, 7), mk_Point(2, 6)
+    )
+
+    private val glider = mk_Population(
+        mk_Point(1, 0), mk_Point(2, 0), mk_Point(3, 0),
+        mk_Point(3, 1), mk_Point(2, 2)
+    )
+
+    @Test
+    @Ignore
+    fun preconditionTests() {
```

Comment:
🚫 This test name is overly generic. In this instance I think it's fine to group together the tests for the `nat1` precondition, but you'll want a more descriptive test name.

I would also place the comment above the `@Test` annotation, rather than in the test body -- the way I read the scope of comments is that they apply to the object immediately below or to the left of the comment. So in this case I was momentarily confused because it looked like the comment applies only to `assertFailsWith<PreconditionFailure> { generations(0, block) }`.